File: CPP/Program/Komputer.cpp
#ifndef KOMPUTER_CPP
#define KOMPUTER_CPP

#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
using namespace std;

class Komputer {
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;
    
public:
    Komputer() {
    }
    
    Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive) {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;
    }
    
    void setNama(string nama) {
        this->nama = nama;
    }
    
    void setCpu(Cpu cpu) {
        this->cpu = cpu;
    }
    
    void setRam(vector<Ram> ramList) {
        this->ramList = ramList;
    }
    
    void setHarddrive(Harddrive harddrive) {
        this->harddrive = harddrive;
    }
    
    void addRam(Ram ram) {
        this->ramList.push_back(ram);
    }
    
    string getNama() {
        return this->nama;
    }
    
    Cpu getCpu() {
        return this->cpu;
    }
    
    vector<Ram> getRamList() {
        return this->ramList;
    }
    
    Harddrive getHarddrive() {
        return this->harddrive;
    }
    
    void displayInfo() {
        cout << "Informasi Komputer:" << endl;
        cout << "Nama : " << this->nama << endl;
        cout << "Cpu : " << this->cpu.getMerk() << ' ' << this->cpu.getNama() 
             << " (" << this->cpu.getJumlahCore() << " Core) " 
             << "~" << this->cpu.getKecepatanGHz() << "GHz" << endl;
        
        cout << "RAM :" << endl;
        for (Ram& ram : this->ramList) {
            cout << "- " << ram.getMerk() << ' ' << ram.getNama() 
                 << " (" << ram.getKapasitasGB() << " GB) " 
                 << ram.getDdr() << endl;
        }
        
        cout << "Harddrive : " << this->harddrive.getTipeDrive() << ' ' 
             << this->harddrive.getMerk() << ' ' << this->harddrive.getNama() 
             << " (" << this->harddrive.getKapasitasGB() << " GB) " << endl;
    }
    
    ~Komputer() {
    }
};

#endif // KOMPUTER_CPP

File: Python/Program/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive

class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, harddrive=None):
        self.__nama = nama
        self.__cpu = cpu if cpu is not None else Cpu()
        self.__ram_list = ram_list if ram_list is not None else []
        self.__harddrive = harddrive if harddrive is not None else Harddrive()
    
    def set_nama(self, nama):
        self.__nama = nama
    
    def set_cpu(self, cpu):
        self.__cpu = cpu
    
    def set_ram(self, ram_list):
        self.__ram_list = ram_list
    
    def set_harddrive(self, harddrive):
        self.__harddrive = harddrive
    
    def add_ram(self, ram):
        self.__ram_list.append(ram)
    
    def get_nama(self):
        return self.__nama
    
    def get_cpu(self):
        return self.__cpu
    
    def get_ram_list(self):
        return self.__ram_list
    
    def get_harddrive(self):
        return self.__harddrive
    
    def display_info(self):
        print("Informasi Komputer:")
        print(f"Nama : {self.__nama}")
        print(f"Cpu : {self.__cpu.get_merk()} {self.__cpu.get_nama()} "
              f"({self.__cpu.get_jumlah_core()} Core) "
              f"~{self.__cpu.get_kecepatan_ghz()}GHz")
        
        print("RAM :")
        for ram in self.__ram_list:
            print(f"- {ram.get_merk()} {ram.get_nama()} "
                  f"({ram.get_kapasitas_gb()} GB) {ram.get_ddr()}")
        
        print(f"Harddrive : {self.__harddrive.get_tipe_drive()} "
              f"{self.__harddrive.get_merk()} {self.__harddrive.get_nama()} "
              f"({self.__harddrive.get_kapasitas_gb()} GB)")